rhp/v2/contracts.go

```diff
@@ -69,52 +69,51 @@ func ContractRenewalCost(fc types.FileContract, contractFee types.Currency) type
-	// data already in the contract, plus their collateral. In the event of a
-	// missed payout, the cost and collateral of the data already in the
-	// contract is subtracted from the host, and sent to the void instead.
-	//
-	// However, it is possible for this subtraction to underflow; this can
-	// happen if baseCollateral is large and MaxCollateral is small. We cannot
-	// simply replace the underflow with a zero, because the host performs the
-	// same subtraction and returns an error on underflow. Nor can we increase
-	// the valid payout, because the host calculates its collateral contribution
-	// by subtracting the contract price and base price from this payout, and
-	// we're already at MaxCollateral. Thus the host has conflicting
-	// requirements, and renewing the contract is impossible until they change
-	// their settings.
-	hostValidPayout := host.ContractPrice.Add(basePrice).Add(totalCollateral)
+// Calculate payouts: the host gets their contract fee, plus the cost of the
+// data already in the contract, plus their collateral. In the event of a
+// missed payout, the cost and collateral of the data already in the
+// contract is subtracted from the host, and sent to the void instead.
+//
+// However, it is possible for this subtraction to underflow; this can
+// happen if baseCollateral is large and MaxCollateral is small. We cannot
+// simply replace the underflow with a zero, because the host performs the
+// same subtraction and returns an error on underflow. Nor can we increase
+// the valid payout, because the host calculates its collateral contribution
+// by subtracting the contract price and base price from this payout, and
+// we're already at MaxCollateral. Thus the host has conflicting
+// requirements, and renewing the contract is impossible until they change
+// their settings.
+func CalculatePayouts(basePrice, baseCollateral, hostCollateral types.Currency, s HostSettings) (types.Currency, types.Currency, types.Currency) {
+	hostValidPayout := s.ContractPrice.Add(basePrice).Add(hostCollateral)
 	voidMissedPayout := basePrice.Add(baseCollateral)
 	if hostValidPayout.Cmp(voidMissedPayout) < 0 {
 		// TODO: detect this elsewhere
 		panic("host's settings are unsatisfiable")
 	}
 	hostMissedPayout := hostValidPayout.Sub(voidMissedPayout)
+	return hostValidPayout, hostMissedPayout, voidMissedPayout
+}
+
+// PrepareContractRenewal constructs a contract renewal transaction.
+func PrepareContractRenewal(currentRevision types.FileContractRevision, renterKey types.PrivateKey, hostKey types.PublicKey, renterPayout, hostCollateral types.Currency, endHeight uint64, host HostSettings, refundAddr types.Address) types.FileContract {
+	// calculate base price and collateral
+	basePrice, baseCollateral := ContractBaseCosts(currentRevision.FileContract, host, endHeight)
+
+	// calculate payouts
+	hostValidPayout, hostMissedPayout, voidMissedPayout := CalculatePayouts(basePrice, baseCollateral, hostCollateral, host)
```

Comment:
I don't think you necessarily need to split `ContractBaseCosts` and `CalculatePayouts` since you always need both together anyway.